File: source/vehicle.cpp
#include <iostream>

#include "vehicle.h"

using namespace std;


Vehicle::Vehicle(cv::Mat image, cv::Rect roi, int class_id)
{
    this->roi = roi;
    this->image = image(roi);
    this->class_id = class_id;
    this->not_found = 0;
}

Vehicle::~Vehicle()
{
    // delete ui;
}

void Vehicle::detect(cv::Mat& image)
{

}

File: python/source/vehicle.py
import cv2
import numpy as np


class Vehicle:
    outputSize = (320, 64)

    def __init__(self, rect, class_id=-1, confidence=0):
        self.rect = rect
        self.confidence = confidence
        self.class_id = class_id
        self.not_found = 0
        self.id = id

    def updateImage(self, frame):
        self.image = self.perspectiveTransform(frame, self.roi)
